pkg/machinepool/autoscaling.go

```diff
@@ -36,7 +36,7 @@ func ensureAutoscalingAnnotations(m mutator.Mutator, machinePool *capiexp.Machin
 		patches = append(patches, *mutator.PatchAdd(fmt.Sprintf("/metadata/annotations/%s", escapeJSONPatchString(annotation.NodePoolMinSize)), fmt.Sprintf("%d", clusterReplicas)))
 	} else {
 		// Parse current value of min Size.
-		min, err := strconv.Atoi(machinePool.Annotations[annotation.NodePoolMinSize])
+		min, err := strconv.ParseInt(machinePool.Annotations[annotation.NodePoolMinSize], 10, 64)
```

Comment:
Not sure if this silences the linter, but this is wrong anyway. The problem that linter is saying is that string is parsed into 64bit variable and then downcast into 32bit one. If linter gets silenced by this, it's plain wrong. This should be like following:
```suggestion
		min, err := strconv.ParseInt(machinePool.Annotations[annotation.NodePoolMinSize], 10, 32)
```

I have fix on its way.

Reply:
thanks

Reply:
What about just using 64-bit numbers here? See my change here https://github.com/giantswarm/azure-admission-controller/pull/210/files#diff-77a4681d42f415cfc2e0ac06004f466a52602a9c1f13a02f4030fedc0a3c2419R30

Reply:
We could use 64bit values for annotation, but since this is about replicas, I figure it would make sense to align with `MachinePool.Spec.Replicas`: https://github.com/kubernetes-sigs/cluster-api/blob/master/exp/api/v1alpha3/machinepool_types.go#L41